packages/alfa-rules/src/common/predicate/is-clipped.ts

```diff
@@ -6,22 +6,26 @@ import { Refinement } from "@siteimprove/alfa-refinement";
 import { Context } from "@siteimprove/alfa-selector";
 import { Style } from "@siteimprove/alfa-style";
 
+import { getOffsetParent } from "@siteimprove/alfa-rules/src/common/expectation/get-offset-parent";
+import { isPositioned } from "./is-positioned";
+
 const { abs } = Math;
 const { isElement } = Element;
-const { or, test } = Predicate;
+const { or, test, not } = Predicate;
 const { and } = Refinement;
 
 const cache = Cache.empty<Device, Cache<Context, Cache<Node, boolean>>>();
 
 /**
  * Checks if a node (or one of its ancestor) is fully clipped
  */
+
 export function isClipped(
   device: Device,
   context: Context = Context.empty()
 ): Predicate<Node> {
-  return (node) =>
-    cache
+  return function isClippedInternal(node: Node): boolean {
```

Comment:
```suggestion
  return function isClipped(node: Node): boolean {
```
We can keep the same name here. It's OK to do so because the outer function returns the inner one, so they are indeed the same thing just with some wrapper around.